src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -102,6 +103,20 @@ function checkAndWarnForMutatedStyle(style1, style2, component) {
   if (style1 == null || style2 == null) {
     return;
   }
+
+  // Remove any NaN values from the style objects before the shallow equal check
+  // so they don't give a false positive
+  for (var key1 in style1) {
+    if (numberIsNaN(style1[key1])) {
+      delete style1[key1];
```

Comment:
@jimfb The shallow-equal check should really use `Object.is`, then this problem should go away naturally.